src/IonParserTextRaw.ts

```diff
@@ -1063,26 +1128,118 @@ export class ParserTextRaw {
     return s;
   }
 
-  private _skip_triple_quote_gap(ii: number, end: number) : number {
-    ii += 2; // skip the two quotes we peeked ahead to see
-    while (ii<end) {
-      let ch: number = this._in.valueAt(ii);
-      if (IonText.is_whitespace(ch)) {
-        // do nothing
-      }
-      else if (ch == CH_SQ) {
-        ii+= 3;
-        if (ii > end || this._in.valueAt(ii-2) != CH_SQ || this._in.valueAt(ii-1) != CH_SQ) {
-          return ii;
-        }
+  private indexWhiteSpace(indice : number, acceptComments : boolean) : number {
+      let ch : number = this._in.valueAt(indice);
+      if(!acceptComments){
+          for(; is_whitespace(ch); ch = this._in.valueAt(indice++)){}
+      } else {
+          for(;is_whitespace(ch) || ch === CH_FORWARD_SLASH;ch = this._in.valueAt(indice++)){
+              if (ch === CH_FORWARD_SLASH) {
+                  ch = this._in.valueAt(indice++);
+                  switch (ch) {
+                      case CH_FORWARD_SLASH:
+                          indice = this.indexToNewLine(indice);
+                          break;
+                      case CH_AS:
+                          indice = this.indexToCloseComment(indice);
+                          break;
+                      default:
+                          indice--;
+                          break;
+                  }
+              }
+          }
       }
-      else {
-        this._error("unexpected character");
+      return indice;
+  }
+
+  private indexToNewLine(indice : number){
```

Comment:
Sorry I could not help myself, the function name is `indexToNewLine` and it takes `indice` plural? :)

Reply:
index is also a verb